File: go_algorithm/numbers/bit/bitwise_or_of_even_numbers_in_an_array.go
package main

import "go_algorithm/test_util"

/*
	https://leetcode.com/problems/bitwise-or-of-even-numbers-in-an-array/

	Cho mảng số nguyên nums. Hãy trả về kết quả phép OR theo từng bit của
	tất cả các số chẵn trong mảng.

	Nếu nums không có số chẵn nào thì trả về 0.
*/
func evenNumberBitwiseORs(nums []int) int {
	res := 0
	for _, v := range nums {
		if v % 2 == 0 {
			res |= v
		}
	}
	return res
}

func main() {
	test_util.AssertInt(0, evenNumberBitwiseORs([]int{7, 9, 11}))
	test_util.AssertInt(6, evenNumberBitwiseORs([]int{1, 2, 3, 4, 5, 6}))
}

File: py-algorithm/number/bit/bitwise_or_of_even_numbers_in_an_array.py
import unittest

from typing import List

"""
    https://leetcode.com/problems/bitwise-or-of-even-numbers-in-an-array/

    Cho một mảng số nguyên nums. Hãy trả về kết quả phép OR theo từng
    bit (bitwise OR) của tất cả các số chẵn trong mảng.

    Nếu trong nums không có số chẵn nào, hãy trả về 0.
"""


class Solution(unittest.TestCase):

    def evenNumberBitwiseORs(self, nums: List[int]) -> int:
        res = 0
        for num in nums:
            if num & 1 == 0:
                res |= num

        return res

    def test_todo(self):
        test_cases = [
            ([1, 2, 3, 4, 5, 6], 6),
            ([7, 9, 11], 0),
        ]
        for arr, expected in test_cases:
            self.assertEqual(expected, self.evenNumberBitwiseORs(arr))


if __name__ == '__main__':
    unittest.main()
